cmd/fleetctl/users_test.go

```diff
@@ -29,3 +29,62 @@ func TestUserDelete(t *testing.T) {
 	assert.Equal(t, "", runAppForTest(t, []string{"user", "delete", "--email", "user1@test.com"}))
 	assert.Equal(t, uint(42), deletedUser)
 }
+
+type notFoundError struct{}
+
+var _ fleet.NotFoundError = (*notFoundError)(nil)
+
+func (e *notFoundError) IsNotFound() bool {
+	return true
+}
+
+func (e *notFoundError) Error() string {
+	return ""
+}
+
+// TestUserCreateForcePasswordReset tests that the `fleetctl user create` command
+// creates a user with the proper "AdminForcePasswordReset" value depending on
+// the passed flags (e.g. SSO users shouldn't be required to do password reset on first login).
+func TestUserCreateForcePasswordReset(t *testing.T) {
+	_, ds := runServerWithMockedDS(t)
+
+	ds.InviteByEmailFunc = func(ctx context.Context, email string) (*fleet.Invite, error) {
+		return nil, &notFoundError{}
+	}
+
+	expectedAdminForcePasswordReset := false
+
+	ds.NewUserFunc = func(ctx context.Context, user *fleet.User) (*fleet.User, error) {
+		assert.Equal(t, expectedAdminForcePasswordReset, user.AdminForcedPasswordReset)
+		return user, nil
+	}
+
+	assert.Equal(t, "", runAppForTest(t, []string{
+		"user", "create",
+		"--email", "foo@example.com",
+		"--name", "foo",
+		"--sso",
+	}))
```

Comment:
To make sure it actually calls NewUserFunc (otherwise it could pass without validating the expected force reset flag), you should add `require.True(t, ds.NewUserFuncInvoked)` after each call and reset the invoked flag to false after.